worlds/hk/__init__.py

```diff
@@ -422,13 +422,15 @@ def set_rules(self):
         player = self.player
         if world.logic[player] != 'nologic':
             goal = world.Goal[player]
-            if goal == Goal.option_siblings:
+            if goal == Goal.option_hollowknight:
+                world.completion_condition[player] = lambda state: state._hk_can_beat_thk(player)
+            elif goal == Goal.option_siblings:
                 world.completion_condition[player] = lambda state: state._hk_siblings_ending(player)
             elif goal == Goal.option_radiance:
                 world.completion_condition[player] = lambda state: state._hk_can_beat_radiance(player)
             else:
-                # Hollow Knight or Any goal.
-                world.completion_condition[player] = lambda state: state._hk_can_beat_thk(player)
+                # Any goal
+                world.completion_condition[player] = lambda state: state._hk_can_beat_thk(player) or state._hk_can_beat_radiance(player)
```

Comment:
Couldn't this make it so that a player who selected THK ending is actually required to fight radiance if dreamers are inaccessible (e.g. on Minimal)

Reply:
The THK ending is covered a few lines above under ```if goal == Goal.option_hollowknight:```.

Reply:
Oh I'm very smart